ecs-init/docker/dependencies.go

```diff
@@ -131,15 +132,18 @@ func (s *_standardFS) ReadFile(filename string) ([]byte, error) {
 }
 
 func isNetworkError(err error) bool {
-	_, ok := err.(*net.OpError)
-	return ok
+	wrapped, isWrapped := err.(*url.Error)
+	if isWrapped {
```

Comment:
was this always a bug or did something change with the returned error from the library which resulted in this change?

Reply:
It appears to me that we were not handling this properly. The Golang stdlib interactions should have been returning the wrapped error variant now handled appropriately. I didn't see anything in the upstream code (in `vendor/`) that suggests that this would have been unwrapped or returned differently. Our test case had a bad assumption (with a hardcoded `netErorr`) and we seemingly built up against that codified assumption.

Reply:
Thanks for that explanation. I wanted to verify that this change doesn't introduce different behavior with different versions of the daemon.